backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -666,22 +667,28 @@ def post(self, request, hardware_id):
         try:
             body = json.loads(request.body)
 
-            origin = body.get("origin", DEFAULT_ORIGIN)
+            post_body = PostBody(**body)
+
+            origin = post_body.origin
             end_datetime = datetime.fromtimestamp(
-                int(body.get('endTimestampInSeconds')),
+                int(post_body.endTimestampInSeconds),
                 timezone.utc
             )
 
             start_datetime = datetime.fromtimestamp(
-                int(body.get('startTimestampInSeconds')),
+                int(post_body.startTimestampInSeconds),
                 timezone.utc
             )
 
-            selected_commits = body.get("selectedCommits", {})
+            selected_commits = post_body.selectedCommits
             self.filterParams = FilterParams(body, process_body=True)
             self.setup_filters()
 
             is_all_selected = len(selected_commits) == 0
+        except ValidationError as e:
+            return HttpResponseBadRequest(
+                getErrorResponseBody(e.errors())
+            )
```

Comment:
Is there a reason why you are using `create_error_response` in backend/kernelCI_app/views/hardwareDetailsCommitHistoryView.py and using this approach in this file?

Reply:
this one is a different error because it is from pydantic